// heroku-cli-exec/commands/copy.js
'use strict';

const child = require('child_process');
const cli = require('heroku-cli-util');
const co = require('co');
const Client = require('ssh2').Client;
const https = require('https')
const url = require('url');
const tty = require('tty');
const path = require('path');
const fs = require('fs');
const stream = require('stream');
const helpers = require('../lib/helpers')
const ssh = require('../lib/ssh')

module.exports = function(topic, command) {
  return {
    topic: topic,
    command: command,
    description: 'Copy a file from a dyno to the local filesystem',
    help: `Usage: heroku ${topic}:${command} FILENAME`,
    args: [ {name: 'file'} ],
    flags: [
      { name: 'dyno', char: 'd', hasValue: true, description: 'specify the dyno to connect to' },
      { name: 'output', char: 'o', hasValue: true, description: 'the name of the output file' }],
    needsApp: true,
    needsAuth: true,
    run: cli.command(co.wrap(run))
  }
};

function * run(context, heroku) {
  cli.error("The heroku-cli-exec plugin has been moved into the core CLI. Please run the following\n" +
            `commands to update your environment and then run ${cli.color.magenta('heroku ps:copy')} again:\n` +
            `${cli.color.magenta('heroku plugins:uninstall heroku-cli-exec')}\n` +
            `${cli.color.magenta('heroku update')}`)
  cli.exit(1)
}
