// heroku-cli-exec/commands/jmap.js
'use strict';

const child = require('child_process');
const cli = require('heroku-cli-util');
const co = require('co');
const Client = require('ssh2').Client;
const https = require('https')
const url = require('url');
const tty = require('tty')
const stream = require('stream')
const uuid = require('uuid');
const helpers = require('../lib/helpers')
const ssh = require('../lib/ssh')

module.exports = function(topic, command) {
  return {
    topic: topic,
    command: command,
    description: 'Generate a heap dump for a Java process',
    help: `Usage: heroku ${topic}:${command}`,
    variableArgs: true,
    flags: [
      { name: 'dyno', char: 'd', hasValue: true, description: 'specify the dyno to connect to' },
      { name: 'hprof', char: 'h', hasValue: false, description: 'Generate a binary heap dump in hprof format' },
      { name: 'output', char: 'o', hasValue: true, description: 'Name of the file to write the dump to' },
    ],
    needsApp: true,
    needsAuth: true,
    run: cli.command(co.wrap(run))
  }
};

function * run(context, heroku) {
  cli.error("The Java commands in the heroku-cli-exec plugin have been moved into the heroku-cli-java plugin.\n" +
            `Please run the following commands to update your environment and then run ${cli.color.magenta('heroku java:jmap')} again:\n` +
            `${cli.color.magenta('heroku update')}\n` +
            `${cli.color.magenta('heroku plugins:uninstall heroku-cli-exec')}\n` +
            `${cli.color.magenta('heroku plugins:install heroku-cli-java')}`);
  cli.exit(1)
}
