// heroku-cli-exec/commands/jvisualvm.js
'use strict';

const child = require('child_process');
const path = require('path');
const cli = require('heroku-cli-util');
const https = require('https')
const http = require('http')
const fs = require('fs')
const co = require('co');
const helpers = require('../lib/helpers')

module.exports = function(topic, command) {
  return {
    topic: topic,
    command: command,
    description: 'Launch VisualVM connected to Heroku',
    help: `Usage: heroku ${topic}:${command}`,
    args: [],
    flags: [{ name: 'dyno', char: 'd', hasValue: true, description: 'specify the dyno to connect to' }],
    needsApp: true,
    needsAuth: true,
    run: cli.command(co.wrap(run))
  }
};

function * run(context, heroku) {
  cli.error("The Java commands in the heroku-cli-exec plugin have been moved into the heroku-cli-java plugin.\n" +
            `Please run the following commands to update your environment and then run ${cli.color.magenta('heroku java:visualvm')} again:\n` +
            `${cli.color.magenta('heroku update')}\n` +
            `${cli.color.magenta('heroku plugins:uninstall heroku-cli-exec')}\n` +
            `${cli.color.magenta('heroku plugins:install heroku-cli-java')}`);
  cli.exit(1)
}
