// heroku-cli-exec/commands/socks.js
'use strict';

const child = require('child_process');
const path = require('path');
const cli = require('heroku-cli-util');
const https = require('https')
const http = require('http')
const fs = require('fs')
const co = require('co');
const helpers = require('../lib/helpers')

module.exports = function(topic, command) {
  return {
    topic: topic,
    command: command,
    description: 'Launch a SOCKS proxy into a dyno',
    help: `Usage: heroku ${topic}:${command}`,
    args: [],
    flags: [{ name: 'dyno', char: 'd', hasValue: true, description: 'specify the dyno to connect to' }],
    needsApp: true,
    needsAuth: true,
    run: cli.command(co.wrap(run))
  }
};

function * run(context, heroku) {
  cli.error("The heroku-cli-exec plugin has been moved into the core CLI. Please run the following\n" +
            `commands to update your environment and then run ${cli.color.magenta('heroku ps:socks')} again:\n` +
            `${cli.color.magenta('heroku plugins:uninstall heroku-cli-exec')}\n` +
            `${cli.color.magenta('heroku update')}`)
  cli.exit(1)
}
